### PR TITLE
Don't close ancestor popovers when fullscreening

### PR DESCRIPTION
**Pull request by josepharhar**
Without this patch, the fullscreening an element inside an open popover will make the fullscreen element display:none.

Issue: https://github.com/whatwg/html/issues/9998
Corresponding HTML PR: https://github.com/whatwg/html/pull/10116

- [x] At least two implementers are interested (and none opposed):
   * Chrome
   * https://bugzilla.mozilla.org/show_bug.cgi?id=1888088
- [x] [Tests](https://github.com/web-platform-tests/wpt) are written and can be reviewed and commented upon at:
   * https://github.com/web-platform-tests/wpt/pull/44146
- [x] [Implementation bugs](https://github.com/whatwg/meta/blob/main/MAINTAINERS.md#handling-pull-requests) are filed:
   * Chromium: https://bugs.chromium.org/p/chromium/issues/detail?id=1520938
   * Gecko: https://bugzilla.mozilla.org/show_bug.cgi?id=1888088
   * WebKit: https://bugs.webkit.org/show_bug.cgi?id=269928
- [x] [MDN issue](https://github.com/whatwg/meta/blob/main/MAINTAINERS.md#handling-pull-requests) is filed: As said in the HTML PR, I'm not sure this edge case warrants any text in MDN
- [x] The top of this comment includes a [clear commit message](https://github.com/whatwg/meta/blob/main/COMMITTING.md) to use. 

(See [WHATWG Working Mode: Changes](https://whatwg.org/working-mode#changes) for more details.)

***
<a href="https://whatpr.org/fullscreen/237.html" title="Last updated on May 7, 2024, 2:21 AM UTC (4bd1b11)">Preview</a> | <a href="https://whatpr.org/fullscreen/237/4cafed1...4bd1b11.html" title="Last updated on May 7, 2024, 2:21 AM UTC (4bd1b11)">Diff</a>

### CODE REVIEW

**Comment by josepharhar** [comment]
This probably won't build until definitions are exported in https://github.com/whatwg/html/pull/10116

**Review by mfreed7 — APPROVED** [review]
LGTM - seems to match Chromium impl, which seems to work.

**Comment by josepharhar** [comment]
@keithamus does your firefox patch also do fullscreen or just dialogs? https://phabricator.services.mozilla.com/D200686

**Comment by mfreed7** [comment]
Side-note, the [equivalent changes](https://github.com/whatwg/html/pull/10116) landed yesterday in HTML for `<dialog>`.

**Comment by keithamus** [comment]
> @keithamus does your firefox patch also do fullscreen or just dialogs? https://phabricator.services.mozilla.com/D200686

Just dialogs. Want me to raise one for fullscreen too?

**Comment by josepharhar** [comment]
@keithamus yes that would be great! then i can link to the bug in the OP

**Comment by keithamus** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=1888088 & https://phabricator.services.mozilla.com/D205802

**Comment by smaug----** [comment]
(uh, popovers are causing troubles everywhere)

**Comment by josepharhar** [comment]
@foolip @domenic could I get a review of this? The corresponding HTML PR has been merged and I filled out the OP

File: fullscreen.bs [hunk 1]
```diff
@@ -67,7 +67,13 @@ is an <a>ordered set</a> of (<a>string</a>, <a>element</a>) <a>tuples</a>. It is
 <p>To <dfn>fullscreen an <var>element</var></dfn>:
 
 <ol>
- <li><p>Run <a>hide all popovers</a> given <var>element</var>'s <a>node document</a>.
+ <li><p>Let <var>hideUntil</var> be the result of running <a>topmost popover ancestor</a> given
+ <var>element</var>, null, and false.
+
+ <li><p>If <var>hideUntil</var> is null, then set <var>hideUntil</var> to <var>element</var>'s
+ <span>node document</span>.</p></li>
```
Comment (domenic) [inline fullscreen.bs:74]:
```suggestion
 <span>node document</span>.
```

**Review by domenic — APPROVED** [review]
LGTM with nit I will fix while merging

**Pull request merged**